### PR TITLE
SessionService, CredentialService, PasswordAbuse

### PR DESCRIPTION
**Pull request by DandilionLau**
* Implement `SessionSevice` and `CredentialService`
* Implement `PasswordAbuseDetection` under `SqlAccount`
* Add unit test for `PasswordAbuseDetection` using `JMockit`
* Use `HMAC256` for JWT encrpytion, can be modified to ES256 later
* Secret for `HMAC256` is hard-coded in `KeyService`, can be switch to SoftHSM later
* Refactor previous code structure according to the spec sheet
* Modify `.scripts` to ensure compatibility
* Now use `JSONObject` for parsing, will switch to `JacksonJSON`

### CODE REVIEW

File: src/main/java/com/yahoo/identity/services/storage/sql/SqlCredentialService.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.yahoo.identity.services.storage.sql;
+
+import com.yahoo.identity.services.credential.Credential;
+import com.yahoo.identity.services.credential.CredentialService;
+
+import javax.annotation.Nonnull;
+
+public class SqlCredentialService implements CredentialService {
```
Comment (sleepingpig) [inline src/main/java/com/yahoo/identity/services/storage/sql/SqlCredentialService.java]: Why the implementation is called SqlCredentialService? It shouldn't need to talk to database.

File: src/main/java/com/yahoo/identity/services/credential/CredentialImpl.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                .withIssuedAt(Date.from(issueTime))
+                .withSubject(subject)
+                .sign(algorithm);
+            return token;
+        } catch (JWTCreationException exception) {
+            throw new BadRequestException("JWT creation does not succeed.");
+        }
+    }
+
+    public String fromString(@Nonnull String credStr) {
```
Comment (sleepingpig) [inline src/main/java/com/yahoo/identity/services/credential/CredentialImpl.java]: The parsing logic of credential string should be done in CredentialService.

File: src/main/java/com/yahoo/identity/services/storage/sql/SqlCredentialService.java (hunk 1 above)
Comment (DandilionLau) [reply src/main/java/com/yahoo/identity/services/storage/sql/SqlCredentialService.java]: > Why the implementation is called SqlCredentialService? It shouldn't need to talk to database.

The current design doesn’t have that file. It is the redundant file brought when rebasing with historical commit. Already deleted.

Comment (DandilionLau) [reply src/main/java/com/yahoo/identity/services/storage/sql/SqlCredentialService.java]: The current design doesn’t have that file. It is the redundant file brought when rebasing with historical commit. Already deleted.

**Pull request closed without merging**